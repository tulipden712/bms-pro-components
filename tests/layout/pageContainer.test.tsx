import { render, mount } from 'enzyme';
import React from 'react';
import type { BasicLayoutProps } from '@ant-design/pro-layout';
import BasicLayout, { PageContainer, FooterToolbar } from '@ant-design/pro-layout';
import { act } from 'react-dom/test-utils';
import { waitForComponentToPaint } from '../util';

describe('PageContainer', () => {
  it('💄 base use', async () => {
    const html = render(<PageContainer title="期贤" />);
    expect(html).toMatchSnapshot();
  });

  it('💄 config is null', async () => {
    const html = render(<PageContainer />);
    expect(html).toMatchSnapshot();
  });

  it('⚡️ support footer', async () => {
    const html = render(
      <PageContainer
        title="期贤"
        footer={[
          <button type="button" key="button">
            right
          </button>,
        ]}
      />,
    );
    expect(html).toMatchSnapshot();
  });

  it('⚡️ support fixedHeader', async () => {
    const html = render(<PageContainer title="期贤" fixedHeader />);
    expect(html).toMatchSnapshot();
  });

  it('⚡️ support fixHeader', async () => {
    const html = render(<PageContainer title="期贤" fixHeader />);
    expect(html).toMatchSnapshot();
  });

  it('⚡️ support loading', async () => {
    const html = render(<PageContainer title="期贤" loading />);
    expect(html).toMatchSnapshot();
  });

  it('🔥 support footer and breadcrumb', async () => {
    const html = render(
      <PageContainer
        title="期贤"
        breadcrumb={{
          routes: [
            {
              path: '/',
              breadcrumbName: 'home',
            },
          ],
        }}
        footer={[
          <button key="right" type="button">
            right
          </button>,
        ]}
      />,
    );
    expect(html).toMatchSnapshot();
  });

  it('🔥 footer bar support extra', async () => {
    const html = render(
      <FooterToolbar
        className="qixian_footer"
        extra={
          <img
            src="https://gw.alipayobjects.com/zos/rmsportal/KDpgvguMpGfqaHPjicRK.svg"
            alt="logo"
          />
        }
      >
        <button key="button" type="button">
          right
        </button>
      </FooterToolbar>,
    );
    expect(html).toMatchSnapshot();
  });

  it('🔥 footer bar support renderContent', async () => {
    const html = render(
      <FooterToolbar
        className="qixian_footer"
        extra={
          <img
            src="https://gw.alipayobjects.com/zos/rmsportal/KDpgvguMpGfqaHPjicRK.svg"
            alt="logo"
          />
        }
        renderContent={() => {
          return 'home_toolbar';
        }}
      >
        <button key="button" type="button">
          right
        </button>
      </FooterToolbar>,
    );
    expect(html).toMatchSnapshot();
  });

  it('🐲 footer should know width', async () => {
    const wrapper = mount<BasicLayoutProps>(
      <BasicLayout>
        <PageContainer
          title="期贤"
          footer={[
            <button type="button" key="button">
              qixian
            </button>,
          ]}
        />
      </BasicLayout>,
    );
    await waitForComponentToPaint(wrapper);

    expect(wrapper?.find('.ant-pro-footer-bar')?.props()?.style?.width).toBe('calc(100% - 208px)');
    wrapper.setProps({
      collapsed: true,
    });

    await waitForComponentToPaint(wrapper);

    expect(wrapper?.find('.ant-pro-footer-bar')?.props()?.style?.width).toBe('calc(100% - 48px)');

    wrapper.setProps({
      layout: 'top',
    });

    expect(wrapper?.find('.ant-pro-footer-bar')?.props()?.style?.width).toBe('100%');
    expect(wrapper.render()).toMatchSnapshot();
  });

  it('🐲 FooterToolbar should know width', async () => {
    const wrapper = mount<BasicLayoutProps>(
      <BasicLayout>
        <PageContainer>
          <FooterToolbar>
            <button type="button" key="button">
              qixian
            </button>
          </FooterToolbar>
        </PageContainer>
      </BasicLayout>,
    );
    await waitForComponentToPaint(wrapper);

    expect(wrapper?.find('.ant-pro-footer-bar')?.props()?.style?.width).toBe('calc(100% - 208px)');
    wrapper.setProps({
      collapsed: true,
    });

    await waitForComponentToPaint(wrapper);

    expect(wrapper.find('.ant-pro-footer-bar')?.props()?.style?.width).toBe('calc(100% - 48px)');

    wrapper.setProps({
      layout: 'top',
    });
    expect(wrapper.find('.ant-pro-footer-bar')?.props()?.style?.width).toBe('100%');
    expect(wrapper.render()).toMatchSnapshot();
    // test useUseEffect render function
    act(() => {
      wrapper.unmount();
    });
  });

  it('🐲 footer is null, do not render footerToolbar ', async () => {
    const wrapper = mount(
      <PageContainer
        footer={[
          <button type="button" key="button">
            qixian
          </button>,
        ]}
      />,
    );
    await waitForComponentToPaint(wrapper);
    expect(wrapper.render()).toMatchSnapshot();

    wrapper.setProps({
      footer: undefined,
    });
    await waitForComponentToPaint(wrapper);
    expect(wrapper.render()).toMatchSnapshot();
  });

  it('🐲 prolayout support breadcrumbProps', async () => {
    const wrapper = mount(
      <BasicLayout
        breadcrumbProps={{
          separator: '>',
          routes: [
            {
              path: 'index',
              breadcrumbName: 'home',
            },
            {
              path: 'first',
              breadcrumbName: 'first',
              children: [
                {
                  path: '/general',
                  breadcrumbName: 'General',
                },
                {
                  path: '/layout',
                  breadcrumbName: 'Layout',
                },
                {
                  path: '/navigation',
                  breadcrumbName: 'Navigation',
                },
              ],
            },
            {
              path: 'second',
              breadcrumbName: 'second',
            },
          ],
        }}
      >
        <PageContainer />
      </BasicLayout>,
    );
    expect(wrapper.render()).toMatchSnapshot();
  });

  it('🐲 header.footer is null, do not render footerToolbar ', async () => {
    const wrapper = mount(
      <PageContainer
        footer={[
          <button type="button" key="button">
            qixian
          </button>,
        ]}
      />,
    );
    await waitForComponentToPaint(wrapper);
    expect(wrapper.find('.ant-pro-footer-bar').exists()).toBeTruthy();

    wrapper.setProps({ footer: undefined });
    await waitForComponentToPaint(wrapper);

    expect(wrapper.find('.ant-pro-footer-bar').exists()).toBeFalsy();
  });

  it('🐲  tabList and onTabChange is run', async () => {
    const fn = jest.fn();
    const wrapper = mount(
      <PageContainer
        title="标题"
        onTabChange={fn}
        tabList={[
          {
            tab: '基本信息',
            key: 'base',
          },
          {
            tab: '详细信息',
            key: 'info',
          },
        ]}
      />,
    );
    await waitForComponentToPaint(wrapper);

    act(() => {
      wrapper.find('.ant-tabs-nav-list .ant-tabs-tab').at(1).simulate('click');
    });

    expect(fn).toBeCalledWith('info');
  });

  it('content is text and title is null', () => {
    const html = render(<PageContainer content="just so so" />);
    expect(html).toMatchSnapshot();

    const html2 = render(<PageContainer extraContent={<div>extraContent</div>} />);
    expect(html2).toMatchSnapshot();
  });
});
