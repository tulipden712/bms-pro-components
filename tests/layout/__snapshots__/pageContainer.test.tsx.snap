// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageContainer content is text and title is null 1`] = `
<div
  class="ant-pro-page-container"
>
  <div
    class="ant-pro-page-container-warp"
  >
    <div
      class="ant-page-header ant-page-header-ghost"
    >
      <div
        class="ant-page-header-content"
      >
        <div
          class="ant-pro-page-container-detail"
        >
          <div
            class="ant-pro-page-container-main"
          >
            <div
              class="ant-pro-page-container-row"
            >
              <div
                class="ant-pro-page-container-content"
              >
                just so so
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="ant-pro-grid-content"
  >
    <div
      class="ant-pro-grid-content-children"
    />
  </div>
</div>
`;

exports[`PageContainer content is text and title is null 2`] = `
<div
  class="ant-pro-page-container"
>
  <div
    class="ant-pro-page-container-warp"
  >
    <div
      class="ant-page-header ant-page-header-ghost"
    >
      <div
        class="ant-page-header-content"
      >
        <div
          class="ant-pro-page-container-detail"
        >
          <div
            class="ant-pro-page-container-main"
          >
            <div
              class="ant-pro-page-container-row"
            >
              <div
                class="ant-pro-page-container-extraContent"
              >
                <div>
                  extraContent
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="ant-pro-grid-content"
  >
    <div
      class="ant-pro-grid-content-children"
    />
  </div>
</div>
`;

exports[`PageContainer ‚ö°Ô∏è support fixHeader 1`] = `
<div
  class="ant-pro-page-container"
>
  <div
    class="ant-pro-page-container-warp"
  >
    <div
      class="ant-page-header ant-page-header-ghost"
    >
      <div
        class="ant-page-header-heading"
      >
        <div
          class="ant-page-header-heading-left"
        >
          <span
            class="ant-page-header-heading-title"
            title="ÊúüË¥§"
          >
            ÊúüË¥§
          </span>
        </div>
      </div>
    </div>
  </div>
  <div
    class="ant-pro-grid-content"
  >
    <div
      class="ant-pro-grid-content-children"
    />
  </div>
</div>
`;

exports[`PageContainer ‚ö°Ô∏è support fixedHeader 1`] = `
<div
  class="ant-pro-page-container"
>
  <div>
    <div
      class=""
    >
      <div
        class="ant-pro-page-container-warp"
      >
        <div
          class="ant-page-header ant-page-header-ghost"
        >
          <div
            class="ant-page-header-heading"
          >
            <div
              class="ant-page-header-heading-left"
            >
              <span
                class="ant-page-header-heading-title"
                title="ÊúüË¥§"
              >
                ÊúüË¥§
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="ant-pro-grid-content"
  >
    <div
      class="ant-pro-grid-content-children"
    />
  </div>
</div>
`;

exports[`PageContainer ‚ö°Ô∏è support footer 1`] = `
<div
  class="ant-pro-page-container"
>
  <div
    class="ant-pro-page-container-warp"
  >
    <div
      class="ant-page-header ant-page-header-ghost"
    >
      <div
        class="ant-page-header-heading"
      >
        <div
          class="ant-page-header-heading-left"
        >
          <span
            class="ant-page-header-heading-title"
            title="ÊúüË¥§"
          >
            ÊúüË¥§
          </span>
        </div>
      </div>
    </div>
  </div>
  <div
    class="ant-pro-grid-content"
  >
    <div
      class="ant-pro-grid-content-children"
    />
  </div>
  <div
    class="ant-pro-footer-bar"
  >
    <div
      class="ant-pro-footer-bar-left"
    />
    <div
      class="ant-pro-footer-bar-right"
    >
      <button
        type="button"
      >
        right
      </button>
    </div>
  </div>
</div>
`;

exports[`PageContainer ‚ö°Ô∏è support loading 1`] = `
<div
  class="ant-pro-page-container"
>
  <div
    class="ant-pro-page-container-warp"
  >
    <div
      class="ant-page-header ant-page-header-ghost"
    >
      <div
        class="ant-page-header-heading"
      >
        <div
          class="ant-page-header-heading-left"
        >
          <span
            class="ant-page-header-heading-title"
            title="ÊúüË¥§"
          >
            ÊúüË¥§
          </span>
        </div>
      </div>
    </div>
  </div>
  <div
    class="ant-pro-grid-content"
  >
    <div
      class="ant-pro-grid-content-children"
    >
      <div
        style="padding-top:100px;text-align:center"
      >
        <div
          class="ant-spin ant-spin-lg ant-spin-spinning"
        >
          <span
            class="ant-spin-dot ant-spin-dot-spin"
          >
            <i
              class="ant-spin-dot-item"
            />
            <i
              class="ant-spin-dot-item"
            />
            <i
              class="ant-spin-dot-item"
            />
            <i
              class="ant-spin-dot-item"
            />
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`PageContainer üê≤ FooterToolbar should know width 1`] = `
<div
  class="ant-design-pro ant-pro-basicLayout screen-md ant-pro-basicLayout-top-menu ant-pro-basicLayout-top"
>
  <section
    class="ant-layout"
    style="min-height: 100%;"
  >
    <div
      class="ant-layout"
      style="position: relative;"
    >
      <header
        class="ant-layout-header ant-pro-top-menu ant-pro-header-dark"
        style="padding: 0px; height: 48px; line-height: 48px; width: 100%; z-index: 19;"
      >
        <div
          class="ant-pro-top-nav-header"
        >
          <div
            class="ant-pro-top-nav-header-main "
          >
            <div
              class="ant-pro-top-nav-header-main-left"
            >
              <div
                class="ant-pro-top-nav-header-logo"
                id="logo"
              >
                <a>
                  <img
                    alt="logo"
                    src="https://gw.alipayobjects.com/zos/antfincdn/PmY%24TNNDBI/logo.svg"
                  />
                  <h1>
                    Ant Design Pro
                  </h1>
                </a>
              </div>
            </div>
            <div
              class="ant-pro-top-nav-header-menu"
              style="flex: 1;"
            />
          </div>
        </div>
      </header>
      <main
        class="ant-layout-content ant-pro-basicLayout-content ant-pro-basicLayout-has-header"
      >
        <div
          class="ant-pro-page-container"
        >
          <div
            class="ant-pro-page-container-warp"
          >
            <div
              class="ant-page-header ant-page-header-ghost"
            >
              <div
                class="ant-page-header-heading"
              >
                <div
                  class="ant-page-header-heading-left"
                >
                  <span
                    class="ant-page-header-heading-title"
                    title="Ant Design Pro"
                  >
                    Ant Design Pro
                  </span>
                </div>
              </div>
            </div>
          </div>
          <div
            class="ant-pro-grid-content"
          >
            <div
              class="ant-pro-grid-content-children"
            >
              <div>
                <div
                  class="ant-pro-page-container-children-content"
                >
                  <div
                    class="ant-pro-footer-bar"
                    style="width: 100%;"
                  >
                    <div
                      class="ant-pro-footer-bar-left"
                    />
                    <div
                      class="ant-pro-footer-bar-right"
                    >
                      <button
                        type="button"
                      >
                        qixian
                      </button>
                    </div>
                  </div>
                </div>
                <div
                  style="height: 48px; margin-top: 24px;"
                />
              </div>
            </div>
          </div>
        </div>
      </main>
    </div>
  </section>
</div>
`;

exports[`PageContainer üê≤ footer is null, do not render footerToolbar  1`] = `
<div
  class="ant-pro-page-container"
>
  <div
    class="ant-pro-grid-content"
  >
    <div
      class="ant-pro-grid-content-children"
    />
  </div>
  <div
    class="ant-pro-footer-bar"
  >
    <div
      class="ant-pro-footer-bar-left"
    />
    <div
      class="ant-pro-footer-bar-right"
    >
      <button
        type="button"
      >
        qixian
      </button>
    </div>
  </div>
</div>
`;

exports[`PageContainer üê≤ footer is null, do not render footerToolbar  2`] = `
<div
  class="ant-pro-page-container"
>
  <div
    class="ant-pro-grid-content"
  >
    <div
      class="ant-pro-grid-content-children"
    />
  </div>
</div>
`;

exports[`PageContainer üê≤ footer should know width 1`] = `
<div
  class="ant-design-pro ant-pro-basicLayout screen-md ant-pro-basicLayout-top-menu ant-pro-basicLayout-top"
>
  <section
    class="ant-layout"
    style="min-height: 100%;"
  >
    <div
      class="ant-layout"
      style="position: relative;"
    >
      <header
        class="ant-layout-header ant-pro-top-menu ant-pro-header-dark"
        style="padding: 0px; height: 48px; line-height: 48px; width: 100%; z-index: 19;"
      >
        <div
          class="ant-pro-top-nav-header"
        >
          <div
            class="ant-pro-top-nav-header-main "
          >
            <div
              class="ant-pro-top-nav-header-main-left"
            >
              <div
                class="ant-pro-top-nav-header-logo"
                id="logo"
              >
                <a>
                  <img
                    alt="logo"
                    src="https://gw.alipayobjects.com/zos/antfincdn/PmY%24TNNDBI/logo.svg"
                  />
                  <h1>
                    Ant Design Pro
                  </h1>
                </a>
              </div>
            </div>
            <div
              class="ant-pro-top-nav-header-menu"
              style="flex: 1;"
            />
          </div>
        </div>
      </header>
      <main
        class="ant-layout-content ant-pro-basicLayout-content ant-pro-basicLayout-has-header"
      >
        <div
          class="ant-pro-page-container"
        >
          <div
            class="ant-pro-page-container-warp"
          >
            <div
              class="ant-page-header ant-page-header-ghost"
            >
              <div
                class="ant-page-header-heading"
              >
                <div
                  class="ant-page-header-heading-left"
                >
                  <span
                    class="ant-page-header-heading-title"
                    title="ÊúüË¥§"
                  >
                    ÊúüË¥§
                  </span>
                </div>
              </div>
            </div>
          </div>
          <div
            class="ant-pro-grid-content"
          >
            <div
              class="ant-pro-grid-content-children"
            />
          </div>
          <div
            class="ant-pro-footer-bar"
            style="width: 100%;"
          >
            <div
              class="ant-pro-footer-bar-left"
            />
            <div
              class="ant-pro-footer-bar-right"
            >
              <button
                type="button"
              >
                qixian
              </button>
            </div>
          </div>
        </div>
      </main>
    </div>
  </section>
</div>
`;

exports[`PageContainer üê≤ prolayout support breadcrumbProps 1`] = `
<div
  class="ant-design-pro ant-pro-basicLayout screen-md ant-pro-basicLayout-side"
>
  <section
    class="ant-layout ant-layout-has-sider"
    style="min-height: 100%;"
  >
    <aside
      class="ant-layout-sider ant-layout-sider-dark ant-pro-sider ant-pro-sider-layout-side"
      style="overflow: hidden; flex: 0 0 208px; max-width: 208px; min-width: 208px; width: 208px;"
    >
      <div
        class="ant-layout-sider-children"
      >
        <div
          class="ant-pro-sider-logo"
          id="logo"
        >
          <a>
            <img
              alt="logo"
              src="https://gw.alipayobjects.com/zos/antfincdn/PmY%24TNNDBI/logo.svg"
            />
            <h1>
              Ant Design Pro
            </h1>
          </a>
        </div>
        <div
          style="flex: 1; overflow-y: auto; overflow-x: hidden;"
        />
        <div
          class="ant-pro-sider-links"
        >
          <ul
            class="ant-menu ant-menu-dark ant-pro-sider-link-menu ant-menu-root ant-menu-inline"
            role="menu"
          >
            <li
              class="ant-menu-item ant-menu-item-only-child ant-pro-sider-collapsed-button"
              role="menuitem"
              style="padding-left: 16px;"
            >
              <span
                aria-label="menu-fold"
                class="anticon anticon-menu-fold"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  data-icon="menu-fold"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M408 442h480c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8H408c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8zm-8 204c0 4.4 3.6 8 8 8h480c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8H408c-4.4 0-8 3.6-8 8v56zm504-486H120c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8zm0 632H120c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8zM115.4 518.9L271.7 642c5.8 4.6 14.4.5 14.4-6.9V388.9c0-7.4-8.5-11.5-14.4-6.9L115.4 505.1a8.74 8.74 0 000 13.8z"
                  />
                </svg>
              </span>
            </li>
          </ul>
        </div>
      </div>
    </aside>
    <div
      class="ant-layout"
      style="position: relative;"
    >
      <header
        class="ant-layout-header ant-pro-fixed-header-action ant-pro-header-dark"
        style="padding: 0px; height: 48px; line-height: 48px; width: 100%; z-index: 19;"
      >
        <div
          class="ant-pro-global-header ant-pro-global-header-layout-side"
        >
          <div
            style="flex: 1;"
          />
        </div>
      </header>
      <main
        class="ant-layout-content ant-pro-basicLayout-content ant-pro-basicLayout-has-header"
      >
        <div
          class="ant-pro-page-container"
        >
          <div
            class="ant-pro-page-container-warp"
          >
            <div
              class="ant-page-header has-breadcrumb ant-page-header-ghost"
            >
              <div
                class="ant-breadcrumb"
              >
                <span>
                  <span
                    class="ant-breadcrumb-link"
                  >
                    <a
                      href="index"
                    >
                      home
                    </a>
                  </span>
                  <span
                    class="ant-breadcrumb-separator"
                  >
                    &gt;
                  </span>
                </span>
                <span>
                  <span
                    class="ant-dropdown-trigger ant-breadcrumb-overlay-link"
                  >
                    <span
                      class="ant-breadcrumb-link"
                    >
                      <a
                        href="first"
                      >
                        first
                      </a>
                    </span>
                    <span
                      aria-label="down"
                      class="anticon anticon-down"
                      role="img"
                    >
                      <svg
                        aria-hidden="true"
                        data-icon="down"
                        fill="currentColor"
                        focusable="false"
                        height="1em"
                        viewBox="64 64 896 896"
                        width="1em"
                      >
                        <path
                          d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                        />
                      </svg>
                    </span>
                  </span>
                  <span
                    class="ant-breadcrumb-separator"
                  >
                    &gt;
                  </span>
                </span>
                <span>
                  <span
                    class="ant-breadcrumb-link"
                  >
                    <a
                      href="second"
                    >
                      second
                    </a>
                  </span>
                  <span
                    class="ant-breadcrumb-separator"
                  >
                    &gt;
                  </span>
                </span>
              </div>
              <div
                class="ant-page-header-heading"
              >
                <div
                  class="ant-page-header-heading-left"
                >
                  <span
                    class="ant-page-header-heading-title"
                    title="Ant Design Pro"
                  >
                    Ant Design Pro
                  </span>
                </div>
              </div>
            </div>
          </div>
          <div
            class="ant-pro-grid-content"
          >
            <div
              class="ant-pro-grid-content-children"
            />
          </div>
        </div>
      </main>
    </div>
  </section>
</div>
`;

exports[`PageContainer üíÑ base use 1`] = `
<div
  class="ant-pro-page-container"
>
  <div
    class="ant-pro-page-container-warp"
  >
    <div
      class="ant-page-header ant-page-header-ghost"
    >
      <div
        class="ant-page-header-heading"
      >
        <div
          class="ant-page-header-heading-left"
        >
          <span
            class="ant-page-header-heading-title"
            title="ÊúüË¥§"
          >
            ÊúüË¥§
          </span>
        </div>
      </div>
    </div>
  </div>
  <div
    class="ant-pro-grid-content"
  >
    <div
      class="ant-pro-grid-content-children"
    />
  </div>
</div>
`;

exports[`PageContainer üíÑ config is null 1`] = `
<div
  class="ant-pro-page-container"
>
  <div
    class="ant-pro-grid-content"
  >
    <div
      class="ant-pro-grid-content-children"
    />
  </div>
</div>
`;

exports[`PageContainer üî• footer bar support extra 1`] = `
<div
  class="qixian_footer ant-pro-footer-bar"
>
  <div
    class="ant-pro-footer-bar-left"
  >
    <img
      alt="logo"
      src="https://gw.alipayobjects.com/zos/rmsportal/KDpgvguMpGfqaHPjicRK.svg"
    />
  </div>
  <div
    class="ant-pro-footer-bar-right"
  >
    <button
      type="button"
    >
      right
    </button>
  </div>
</div>
`;

exports[`PageContainer üî• footer bar support renderContent 1`] = `
<div
  class="qixian_footer ant-pro-footer-bar"
>
  home_toolbar
</div>
`;

exports[`PageContainer üî• support footer and breadcrumb 1`] = `
<div
  class="ant-pro-page-container"
>
  <div
    class="ant-pro-page-container-warp"
  >
    <div
      class="ant-page-header has-breadcrumb ant-page-header-ghost"
    >
      <div
        class="ant-breadcrumb"
      >
        <span>
          <span
            class="ant-breadcrumb-link"
          >
            <span>
              home
            </span>
          </span>
          <span
            class="ant-breadcrumb-separator"
          >
            /
          </span>
        </span>
      </div>
      <div
        class="ant-page-header-heading"
      >
        <div
          class="ant-page-header-heading-left"
        >
          <span
            class="ant-page-header-heading-title"
            title="ÊúüË¥§"
          >
            ÊúüË¥§
          </span>
        </div>
      </div>
    </div>
  </div>
  <div
    class="ant-pro-grid-content"
  >
    <div
      class="ant-pro-grid-content-children"
    />
  </div>
  <div
    class="ant-pro-footer-bar"
  >
    <div
      class="ant-pro-footer-bar-left"
    />
    <div
      class="ant-pro-footer-bar-right"
    >
      <button
        type="button"
      >
        right
      </button>
    </div>
  </div>
</div>
`;
